### ISSUE TITLE
The speed of training DeepDPM based on pretrained embeddings  is very slow 

### ISSUE THREAD

**Issue by Pro-xiaowen** [BGU-CS-VIL/DeepDPM#8]
In learning clustering using DeepDPM.py based on my data's embedding, the time cost of one epoch is 1h,  my feature dim is 143 , and the number of my data is 10k. In the training , i follow your training setting in  DeepDPM.py, expect "latent_dim","prior_nu" .

My training  command as follows: 
python DeepDPM.py --dataset renti \
--dir ./pretrained_embeddings/barlowtwins \
--latent_dim 147 \
--prior_nu 150 \
--prior_sigma_scale 0.05 \
--offline \
--gpus 2,3,5,6 \
--offline \
--exp_name 'renti_pca0.95_officeparam_real_0424' \
--max_epochs 1000

Could you help me find this abnormal problem?
Thanks~

**Comment by meitarronen** [comment]
Hi,
One hour per epoch is unusual.
It is only for the first epoch? Or every epoch?
Could you please provide us with some logging so we can help understand what is going on?
Also, could you try with only one GPU and make sure that it is used? (This information is printed at the beginning of the training, e.g., number of the available GPUs and the number of used GPUs)

**Comment by Pro-xiaowen** [comment]
> Hi, One hour per epoch is unusual. It is only for the first epoch? Or every epoch? Could you please provide us with some logging so we can help understand what is going on? Also, could you try with only one GPU and make sure that it is used? (This information is printed at the beginning of the training, e.g., number of the available GPUs and the number of used GPUs)
The time cost of one epoch is 15min when I trained using only one gpu,  which is this normal? the logs as following:


 - Epoch 9:   6%|███████▋                                                                                                                         | 660/11039 [01:04<16:46, 10.31it/s, loss=0, v_num=^Epoch 9:  13%|████████████████▏                                                                                                               | 1398/11039 [02:04<14:16, 11.26it/s, loss=0, v_num=]Epoch 9:  19%|████████████████████████▍                                                                                                       | 2112/11039 [03:03<12:57, 11.48it/s, loss=0, v_num=^Epoch 9:  26%|█████████████████████████████████                                                                                               | 2856/11039 [04:04<11:39, 11.70it/s, loss=0, v_num=^Epoch 9:  32%|█████████████████████████████████████████▎                                                                                      | 3564/11039 [05:04<10:38, 11.72it/s, loss=0, v_num=]Epoch 9:  39%|█████████████████████████████████████████████████▋                                                                              | 4290/11039 [06:03<09:32, 11.79it/s, loss=0, v_num=^Epoch 9:  46%|██████████████████████████████████████████████████████████▎                                                                     | 5028/11039 [07:03<08:26, 11.86it/s, loss=0, v_num=]Epoch 9:  52%|██████████████████████████████████████████████████████████████████▋                                                             | 5754/11039 [08:03<07:24, 11.89it/s, loss=0, v_num=^Epoch 9:  59%|███████████████████████████████████████████████████████████████████████████▊                                                    | 6534/11039 [09:04<06:15, 12.01it/s, loss=0, v_num=]Epoch 9:  66%|████████████████████████████████████████████████████████████████████████████████████▏                                           | 7260/11039 [10:04<05:14, 12.02it/s, loss=0, v_num=]Epoch 9:  73%|█████████████████████████████████████████████████████████████████████████████████████████████                                   | 8022/11039 [11:04<04:09, 12.08it/s, loss=0, v_num=^Epoch 9:  79%|█████████████████████████████████████████████████████████████████████████████████████████████████████▋                          | 8772/11039 [12:03<03:07, 12.12it/s, loss=0, v_num=^Epoch 9:  86%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████▌                 | 9540/11039 [13:03<02:03, 12.17it/s, loss=0, v_num=^Epoch 9:  93%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▎        | 10284/11039 [14:03<01:01, 12.19it/s, loss=0, v_num=^Epoch 9: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉| 11034/11039 [15:04<00:00, 12.20it/s, loss=0, v_num=^@alidating: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉| 10948/10953 [14:47<00:00, 13.33it/sEvaluating...100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉| 10952/10953 [14:47<00:00,  9.61it/s]
- [ ] > - Evaluating...

**Comment by meitarronen** [comment]
15 min per epoch is still unusual...
What is the size of the dataset (train and validation)?
Could you also please try to use the code with "--log_metrics_at_train False" and "--log_emb never" to make sure the time consumption isn't due to the evaluation or plotting?

**Comment by Pro-xiaowen** [comment]
已收到祝好许晓文

**Comment by Treeboy2762** [comment]
Hi,

Did you use a large image dataset? I am experiencing the same problem, and I suspect it is the scale of the dataset that causes the issue.

**Comment by Pro-xiaowen** [comment]
已收到祝好许晓文

**Comment by meitarronen** [comment]
Hi, 
@Treeboy2762 we have trained on the entire ImageNet dataset. It took approximately 5 days, which is comparable with SPTA methods.
However, on MNIST it shouldn't take more than 8 hrs even on a weak GPU.

@Pro-xiaowen could you please share the size of your dataset?

**Comment by meitarronen** [comment]
> 已收到祝好许晓文

I'm sorry II do not understand this. Please write either in English or Hebrew (:

**Comment by Treeboy2762** [comment]
It makes much sense now. Thank you for your detailed reply!

+ He means "received, much thanks" with his name. I suspect it's auto-reply?

**Comment by meitarronen** [comment]
@Treeboy2762 Thank you!
Closing this issue for now then
